lib/msf/core/exploit/remote/smb/client/microsoft_kerberos_authenticator.rb

```diff
@@ -0,0 +1,196 @@
+# -*- coding: binary -*-
+
+#
+# This class acts as standalone authenticator for Microsoft Kerberos and SMB
+#
+
+class Msf::Exploit::Remote::SMB::Client::MicrosoftKerberosAuthenticator
+  extend Forwardable
+  include Msf::Exploit::Remote::Kerberos::Client
+
+  attr_reader :domain, :username, :password, :hostname, :host, :port, :timeout, :framework, :framework_module
+
+  def_delegators :@framework_module,
+                :print_status,
+                :store_loot
+
+  def initialize(domain: nil, username: nil, password: nil, hostname: nil, host: nil, port: 88, timeout: nil, framework: nil, framework_module: nil)
```

Comment:
Should timeout just default to 25 or something else in case there's not a datastore option to set it?